app/assets/javascripts/discourse/app/components/sidebar/api-section.hbs

```diff
@@ -1,45 +1,46 @@
-<Sidebar::Section
-  @sectionName={{this.section.name}}
-  @headerLinkText={{this.section.text}}
-  @headerLinkTitle={{this.section.title}}
-  @headerActionsIcon={{this.section.actionsIcon}}
-  @headerActions={{this.section.actions}}
-  @willDestroy={{this.section.willDestroy}}
-  @collapsable={{@collapsable}}
-  @displaySection={{this.section.displaySection}}
-  @hideSectionHeader={{this.section.hideSectionHeader}}
->
-
-  {{#each this.section.links as |link|}}
-    <Sidebar::SectionLink
-      @linkName={{link.name}}
-      @linkClass={{link.classNames}}
-      @route={{link.route}}
-      @model={{link.model}}
-      @models={{link.models}}
-      @href={{link.href}}
-      @title={{link.title}}
-      @contentCSSClass={{link.contentCSSClass}}
-      @prefixColor={{link.prefixColor}}
-      @prefixBadge={{link.prefixBadge}}
-      @prefixType={{link.prefixType}}
-      @prefixValue={{link.prefixValue}}
-      @prefixCSSClass={{link.prefixCSSClass}}
-      @suffixType={{link.suffixType}}
-      @suffixValue={{link.suffixValue}}
-      @suffixCSSClass={{link.suffixCSSClass}}
-      @hoverType={{link.hoverType}}
-      @hoverValue={{link.hoverValue}}
-      @hoverAction={{link.hoverAction}}
-      @hoverTitle={{link.hoverTitle}}
-      @currentWhen={{link.currentWhen}}
-      @didInsert={{link.didInsert}}
-      @willDestroy={{link.willDestroy}}
-      @content={{link.text}}
-      @contentComponent={{component
-        link.contentComponent
-        status=link.contentComponentArgs
-      }}
-    />
-  {{/each}}
-</Sidebar::Section>
\ No newline at end of file
+{{#if this.shouldDisplay}}
```

Comment:
It's super fast ATM doing this, so it's not necessary, but have you tried adding a css class that does display:none instead of adding/removing elements as David suggested? Just curious.

Reply:
No, once it became fast enough, I stopped improving